### DISCUSSION TITLE
Type 'string | undefined' is not assignable to type 'string'

### DISCUSSION THREAD

**Discussion by dedku** [keystonejs/keystone#7508] [Questions] [answered]
Hi, im new to keystonejs.
I have encountered this wired  problem error "Type 'string | undefined' is not assignable to type 'string'.
  Type 'undefined' is not assignable to type 'string'" here : 
```
import { statelessSessions } from '@keystone-6/core/session'
const session = statelessSessions({
  secret: process.env.SESSION_SECERT,
});
```
same goes here:
 [https://github.com/JedWatson/prisma-day-keystone-workshop/blob/main/schema/content.ts](url)
when i clone this project, everything is working fine, but when i copy defaultSlug function to my project (yarn create keystone-app) and assign it to defaultValue in slug, i have a "Type '({ context, originalInput }: any) => string' is not assignable to type 'string'" error.

ts config : 

```
{
  "compilerOptions": {
    "target": "esnext",
    "module": "commonjs",
    "strict": true,
    "noEmit": true,
    "esModuleInterop": true,
    "forceConsistentCasingInFileNames": true,
    "allowJs": false,
    "moduleResolution": "node",
    "resolveJsonModule": true,
    "isolatedModules": true,
    "skipLibCheck": true
  }
}
```

Is there a problem with my config? or am i missing something?

**Comment by raveling** [comment, marked as answer]
@dedku the error is correct, you should check that the environment variable is set and throw an error if it isn't
